### ISSUE TITLE
NewJobWithID needed?

### ISSUE THREAD

**Issue by Fishwaldo** [sherifabdlnaby/sched#5]
Just seems redundant as the Only Caller is NewJob?

https://github.com/sherifabdlnaby/sched/blob/6eeb2cc626e821489a11ee8219b239ced2f900a2/job/job.go#L32

**Comment by sherifabdlnaby** [comment]
Yes, it was there so that maybe this sub package would be imported in an external project as a standalone package. But actually i am considering making the Job package not exported all together so we don’t have to maintain its backward compatibility.
